BayesianTools/vignettes/BayesianTools.Rmd

```diff
@@ -213,9 +213,103 @@ BS <- createBayesianSetup(likelihood, parallel = "external", ...)
+
+
+#### 3. Multi-core and cluster calculations
+If you want to do your calculations on a cluster there are several ways to achieve it.
+
+In the first case you want to parallize n internal (not overall chains) on n cores. The argument "parallel = T" in "createBayesianSetup" allows only at most parallelization on 3 cores for the SMC, DEzs and DreamsSamplers. But by setting "parallel = n" to n cores in the "createBayesianSetup" and giving the settings list a "startValue" matrix with n rows, the internal chains of DEzs and DREAMzs will be parallelized on n cores. This works only for the DEzs and DREAMzs samplers.
+
+```{r, eval = FALSE}
+## n = Number of cores
+x <- c(1:10)
+likelihood <- function(param) return(sum(dnorm(x, mean = param, log = T)))
+bayesianSetup <- createBayesianSetup(likelihood, parallel = n, lower = -5, upper = 5)
+
+## give runMCMC a matrix with n rows of proposals as startValues or sample n times from the previous created sampler
+out <- runMCMC(bayesianSetup, settings = list(iterations = 100000, startValues = bayesianSetup$prior$sampler(n)))
+
+```
+
+In the second case you want to parallize n internal chains on n cores with a external parallilzed likelihood function. Unlike the previous case, that way DEzs, DREAMzs, and SMC samplers can be paralllized. 
+
+```{r, eval = FALSE}
+### Create Cluster with n Cores
+cl <- parallel::makeCluster(n)
+
+## Definition of the likelihood
+likelihood  <- function(X) sum(dnorm(c(1:10), mean = X, log = T))
+
+## Definition of the likelihood which will be calculated in parallel. Instead of the parApply function, we could also define a costly parallelized likelihood
+Plikelihood <- function(param) parallel::parApply(cl = cl, X = param, MARGIN = 1, FUN = likelihood)
+
+## export functions, dlls, libraries
+# parallel::clusterEvalQ(cl, library(BayesianTools))
+parallel::clusterExport(cl, varlist = list(likelihood))
+
+## create BayesianSetup
+bayesianSetup <- createBayesianSetup(Plikelihood, lower = -10, upper = 10, parallel = 'external')
+
+## For this case we want to parallelize the internal chains, therefore we create a n row matrix with startValues, if you parallelize a model in the likelihood, do not set a n*row Matrix for startValue
+settings = list(iterations = 100, nrChains = 1, startValue = bayesianSetup$prior$sampler(n))
+
+## runMCMC
+out <- runMCMC(bayesianSetup, settings, sampler = "DEzs")
+```
+
+In a another case your likelihood requires a costly model which is parallized. Start your cluster and export your model, the required libraries, and dlls. Now you can start your calculations with the argument "parallel = external" in createBayesianSetup.
+
+```{r, eval = FALSE}
+### Create Cluster with n Cores
+cl <- parallel::makeCluster(n)
+
+## export your model
+# parallel::clusterExport(cl, varlist = list(complexModel))
+
+## Definition of the likelihood
+likelihood  <- function(param) {
+  # ll <- complexModel(param)
+  # return(ll)
+} 
+
+## create BayesianSetup and settings
```

Comment:
you are sometimes writing BayesianSetup and sometimes Bayesian Setup. Maybe unify this.